### ISSUE TITLE
Redesign the default trunck size in `from_binary`

### ISSUE THREAD

**Issue by rogerkuou** [MotionbyLearning/sarxarray#3]
In [_io.py](https://github.com/MotionbyLearning/sarxarray/blob/main/sarxarray/_io.py), when loading data by `from_binary`, a default 50-row chunking is applied. This chunk size needs to be based more reasonably, preferably on the pixel size of SLC. This means we need to reimplement the chunking to 2-D. Besides, in the current implementation, the trunk reading function is concatenated in a loop in `_mmap_dask_array`, the efficiency can be improved by applying this in a vector operation.

One can work on this on Spider with [this notebook](https://github.com/MotionbyLearning/sarxarray/blob/3_chunk_size/examples/demo_sarxarray_sprider.ipynb). The following cell should be faster:
```python
stack = sarxarray.from_binary(list_slcs, shape, dtype=dtype)
stack
```

To do:

- [x] Add 2D chunking argument
- [ ] Vectorization of chunk concatenation

**Comment by rogerkuou** [comment]
When testing in a larger dataset azimuth: 10018 range: 68656 time: 375) on Spider, `from_binary` takes ~1min to finish. We can think of doing vectorized operations to increase the speed.

**Comment by fnattino** [comment]
Hi @rogerkuou  and @cpranav93, a Dask array can be constructed from the list of delayed objects that correspond to the individual chunks as described [here](https://docs.dask.org/en/stable/array-design.html) (see example on the [eye function](https://docs.dask.org/en/stable/array-design.html#example-eye-function)). This could be a way to implement the 2D chunking and might be more efficient because it works at a lower level in Dask - but this should actually be tested.

**Comment by rogerkuou** [comment]
Fixed and merged in #14
